serde/src/de/impls.rs

```diff
@@ -131,6 +131,54 @@ macro_rules! impl_deserialize_num {
     };
 }
 
+macro_rules! impl_deserialize_saturating_num {
+    ($primitive:ident, $deserialize:ident ) => {
+        #[cfg(not(no_core_num_saturating))]
+        impl<'de> Deserialize<'de> for Saturating<$primitive> {
+            #[inline]
+            fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+            where
+                D: Deserializer<'de>,
+            {
+                struct SaturatingVisitor;
+
+                fn convert_saturating<T: core::convert::TryInto<$primitive> + Default + Ord + Copy>(
+                    v: T,
+                ) -> Saturating<$primitive> {
+                    // We are taking advantage of T::Default being 0 for all numeric types
+                    // (except for NonZero<T> which we ignore here)
+                    Saturating(v.try_into().unwrap_or_else(|_| {
+                        if v < T::default() {
+                            $primitive::MIN
+                        } else {
+                            $primitive::MAX
+                        }
+                    }))
```

Comment:
you could inline this directly into `visit_saturating` by using `core::convert::TryFrom::<$ty>::try_from(v)`. That way you can also use `0`.